aim/web/ui_v2/src/utils/encoder/streamEncoding.ts

```diff
@@ -0,0 +1,270 @@
+// @ts-nocheck
+
+import { isEqual } from 'lodash-es';
+import struct from '@aksel/structjs';
+
+const PATH_SENTINEL = 0xfe;
+const SIZE_T = 'q';
+
+const NONE = 0;
+const BOOL = 1;
+const INT = 2; // Stored as 64-bit Long Long
+const FLOAT = 3; // Stored as 64-bit Double
+const STRING = 4;
+const BYTES = 5;
+const ARRAY = 6;
+const OBJECT = 7;
+
+const ArrayFlag = {};
+const ObjectFlag = {};
+
+function decodeNone(buffer: ArrayBuffer) {
+  return null;
+}
+
+function decodeBool(buffer: ArrayBuffer) {
+  return struct('?').unpack(buffer)[0];
+}
+
+function decodeInt(buffer: ArrayBuffer) {
+  const len = buffer.byteLength;
+  if (len === struct('q').size) {
+    return struct('q').unpack(buffer)[0];
+  } else if (len === struct('l').size) {
+    return struct('l').unpack(buffer)[0];
+  } else if (len === struct('h').size) {
+    return struct('h').unpack(buffer)[0];
+  }
+}
+
+function decodeFloat(buffer: ArrayBuffer) {
+  const len = buffer.byteLength;
+  if (len === struct('d').size) {
```

Comment:
It turns out @aksel/structjs uses BigEndian by default (God knows why). Because of this we need to prepend all the format-strings with '<' for little-endianness.

I will make sure to add explicit little-endianness to python side as well.